google/cloud/bigtable/examples/bigtable_samples_instance_admin_ext.cc

```diff
@@ -0,0 +1,305 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/bigtable/instance_admin.h"
+#include "google/cloud/bigtable/instance_admin_client.h"
+#include <google/protobuf/text_format.h>
+#include <typeindex>
+#include <typeinfo>
+
+namespace btproto = ::google::bigtable::admin::v2;
+
+namespace {
+struct Usage {
+  std::string msg;
+};
+
+char const* ConsumeArg(int& argc, char* argv[]) {
+  if (argc < 2) {
+    return nullptr;
+  }
+  char const* result = argv[1];
+  std::copy(argv + 2, argv + argc, argv + 1);
+  argc--;
+  return result;
+}
+
+void PrintUsage(int argc, char* argv[], std::string const& msg) {
+  std::string const cmd = argv[0];
+  auto last_slash = std::string(cmd).find_last_of('/');
+  auto program = cmd.substr(last_slash + 1);
+  std::cerr << msg << "\nUsage: " << program
+            << " <command> <project_id> [arguments]\n\n"
+            << "Examples:\n";
+  for (auto example :
+       {"run my-project my-instance my-cluster us-central1-f",
+        "create-dev-instance my-project my-instance us-central1-f",
+        "delete-instance my-project my-instance",
+        "create-cluster my-project my-instance my-cluster us-central1-a",
+        "delete-cluster my-project my-instance my-cluster"}) {
+    std::cerr << "  " << program << " " << example << "\n";
+  }
+  std::cerr << std::flush;
+}
+
+// This full example demonstrate various instance operations
+// by initially creating instance of type PRODUCTION
+void RunInstanceOperations(std::string project_id, int argc, char* argv[]) {
+  if (argc != 4) {
+    throw Usage{"run: <project-id> <instance-id> <cluster-id> <zone>"};
+  }
+  google::cloud::bigtable::InstanceId instance_id(ConsumeArg(argc, argv));
+  google::cloud::bigtable::ClusterId cluster_id(ConsumeArg(argc, argv));
+  std::string const zone = ConsumeArg(argc, argv);
+
+  // [START connect_instance_admin]
+  google::cloud::bigtable::InstanceAdmin instance_admin(
+      google::cloud::bigtable::CreateDefaultInstanceAdminClient(
+          project_id, google::cloud::bigtable::ClientOptions()));
+  // [END connect_instance_admin]
+
+  // [START bigtable_check_instance_exists]
+  std::cout << "\nCheck Instance exists: " << std::endl;
+  auto instances = instance_admin.ListInstances();
+  auto instance_name =
+      instance_admin.project_name() + "/instances/" + instance_id.get();
+  bool instanceExists =
```

Comment:
The naming conventions for C++ is `instance_exists` please fix.

Reply:
fixed.